src/components/menu-surface/menu-surface.scss

```diff
@@ -10,7 +10,17 @@
 @include menu.core-styles;
 
 .mdc-menu-surface {
-    width: var(--menu-surface-width, auto);
     max-height: 100%;
     position: relative;
+    --mdc-menu-max-width: var(
+        --menu-surface-width,
+        min(calc(100vw - 2rem), 20rem)
+    );
+}
+
+:host(limel-menu-surface.has-grid-layout) {
+    .mdc-menu-surface {
+        width: var(--menu-surface-width, min(calc(100vw - 2rem), 40rem));
+        max-width: unset;
```

Comment:
Regarding previous comment and this one, you're right! A name like `--menu-max-width` would have been more semantically correct and definitely a better name. But we were afraid someone may have used this variable (`--menu-surface-width`) in their code. Because menus are so widely used in all solutions. 

Of course one way would be deprecating the previous variable before going forward. But that comes with its pains, and would be a big breaking change.

Additionally, the usage of this custom variable in the previous code was not so nice neither. Because it was put on the `width` property. It would mean that any value that consumers would put there, would give the menus a hard-coded and static width. 

So even if their menu item labels were too short (lets say in their English solution versus Finish), the menu component would show them a lot of dead space. Basically a wide menu with short text labels.

Then we thought we could actually use the custom variable (ignoring the lack of perfection in its name) and put it in the `--mdc-menu-max-width` instead, and combine it with a `min` function. This way, we would make sure menus will remain responsive, even if the consumer has put a static value for the `--menu-surface-width`. Because now, we are applying it to the `max-width` property, instead of the `width` property.


I hope the story of our decisions is more clear now. Otherwise just ask or let's have a live talk :)

What do you suggest? Should we depreciate `--menu-surface-width` and rename it to `--menu-max-surface-width`? Or is this one OK?

Reply:
> Edit: Oh, and we even unset the max-width… 🤔

This is because we are saying `width: auto;` and `max-width: the fancy variabel stuff…` on normal menus. 

While in grid menus, we'd need a static (but still responsive) width, otherwise they will not render in a grid mode. The static `width` will need to use the `min` function to become responsive. And if we inherit the `max-width` from the normal menus, any value used for  `width` that is larger than that inherited `max-width` will not work. 

It's a bit of a mess conceptually, but I think we have handled it well. And believe me, we ourselves were totally 🧠 💨ing the whole way 😄

Reply:
> > Edit: Oh, and we even unset the max-width… 🤔
> 
> This is because we are saying `width: auto;` and `max-width: the fancy variabel stuff…` on normal menus.
> 
> While in grid menus, we'd need a static (but still responsive) width, otherwise they will not render in a grid mode. The static `width` will need to use the `min` function to become responsive. And if we inherit the `max-width` from the normal menus, any value used for `width` that is larger than that inherited `max-width` will not work.
> 
> It's a bit of a mess conceptually, but I think we have handled it well. And believe me, we ourselves were totally 🧠 💨ing the whole way 😄

Ah! Ok, then I understand 👍

> Regarding previous comment and this one, you're right! A name like `--menu-max-width` would have been more semantically correct and definitely a better name. But we were afraid someone may have used this variable (`--menu-surface-width`) in their code. Because menus are so widely used in all solutions.
> 
> Of course one way would be deprecating the previous variable before going forward. But that comes with its pains, and would be a big breaking change.
> 
> […snip…]
> 
> I hope the story of our decisions is more clear now. Otherwise just ask or let's have a live talk :)
> 
> What do you suggest? Should we depreciate `--menu-surface-width` and rename it to `--menu-max-surface-width`? Or is this one OK?

I understand and agree with the reasoning 👍

Deprecating the variable is not the same as removing it. When deprecating, we make both the old and the new variable name work, with the new taking precedence if both are specified. We also mark the old as deprecated in documentation, and use the new one in any examples. Then the deprecated variable can eventually be removed in some future version, but _that_ would indeed be a breaking change.

I don't think we should complicate _this_ PR further by adding the changes necessary for deprecating the variable. We can do that in a separate PR if we want to.